bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandBaseUnitHandler.java

```diff
@@ -0,0 +1,169 @@
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.touchwand.internal.dto.TouchWandUnitData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonParser;
+
+/**
+ * The {@link TouchWandBaseUnitHandler} is responsible for handling commands and status updates
+ * for TouchWand units. This is an abstract class , units should implement the specific command
+ * handling and status updates.
+ *
+ * @author Roie Geron - Initial contribution
+ *
+ */
+@NonNullByDefault
+public abstract class TouchWandBaseUnitHandler extends BaseThingHandler implements TouchWandUnitUpdateListener {
+
+    public TouchWandBaseUnitHandler(Thing thing) {
+        super(thing);
+    }
+
+    @NonNullByDefault({})
+    protected String unitId;
+    protected final Logger logger = LoggerFactory.getLogger(TouchWandBaseUnitHandler.class);
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    @NonNullByDefault({})
+    protected TouchWandBridgeHandler bridgeHandler;
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = new HashSet<>(
+            Arrays.asList(THING_TYPE_SHUTTER, THING_TYPE_SWITCH, THING_TYPE_WALLCONTROLLER, THING_TYPE_DIMMER));
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            // updateTouchWandUnitState(getUnitState(unitId));
+        } else {
+            touchWandUnitHandleCommand(command);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Handler disposed.");
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+        }
+        if (bridgeHandler != null) {
+            bridgeHandler.unregisterUpdateListener(this);
+        }
+    }
+
+    // @SuppressWarnings("null")
+    @Override
+    public void initialize() {
+        ThingStatus bridgeStatus;
+
+        Bridge bridge = getBridge();
+        if (bridge == null || !(bridge.getHandler() instanceof TouchWandBridgeHandler)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            logger.warn("Trying to initialize {} without a bridge", getThing().getUID());
+            return;
+        }
+
+        bridgeHandler = (TouchWandBridgeHandler) bridge.getHandler();
+
+        bridgeStatus = bridge.getStatus();
+
+        unitId = getThing().getUID().toString().split(":")[3]; // touchwand id
```

Comment:
Actually I meant this:
```suggestion
        unitId = getThing().getUID().getId();
```

Reply:
yep , much more elegant .
I changed it and commited